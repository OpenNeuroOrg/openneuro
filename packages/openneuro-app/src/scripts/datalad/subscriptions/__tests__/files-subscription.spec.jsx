import React from 'react'
import { shallow } from 'enzyme'
import FilesSubscription, {
  deleteFilesReducer,
} from '../files-subscription.jsx'

describe('FilesSubscription', () => {
  it('renders with common props', () => {
    const wrapper = shallow(<FilesSubscription datasetId="ds001" />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('deleteFilesReducer', () => {
  let draft, filesToDelete
  beforeEach(() => {
    draft = {
      a: 1,
      files: [
        { filename: 'deleted/file.txt' },
        { filename: 'another_deleted_file.txt' },
        { filename: 'path/to/file.txt' },
        { filename: 'deleted/file/again/here' },
      ],
      steak: 'sauce',
    }
    filesToDelete = [
      { filename: 'deleted:file.txt' },
      { filename: 'another_deleted_file.txt' },
      { filename: 'deleted:file:again:here' },
    ]
  })
  it('removes files from draft', () => {
    const output = deleteFilesReducer(filesToDelete, draft)
    expect(output).toEqual({
      a: 1,
      files: [{ filename: 'path/to/file.txt' }],
      steak: 'sauce',
    })
  })
})
