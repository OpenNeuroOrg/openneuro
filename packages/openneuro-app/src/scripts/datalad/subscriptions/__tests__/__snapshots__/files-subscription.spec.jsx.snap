// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilesSubscription renders with common props 1`] = `
<Subscription
  onSubscriptionData={[Function]}
  subscription={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "filesUpdated",
          },
          "operation": "subscription",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "datasetId",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "datasetId",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "filesUpdated",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "action",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "payload",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "filename",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "size",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "datasetId",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 178,
        "start": 0,
      },
    }
  }
  variables={
    Object {
      "datasetId": "ds001",
    }
  }
/>
`;
